src/commands/watch.ts

```diff
@@ -0,0 +1,114 @@
+        code: flags.string({ char: 'c', dependsOn: ['username'], description: '2FA code of the user' }),
+        i2fa: flags.boolean({ description: 'interactively ask for 2FA code' }),
+    };
+
+    public async run() {
+
+        const { flags } = this.parse(Watch);
+        const fd = new FolderDetails(this);
+        try {
+            await fd.readInfoFile();
+        } catch (e) {
+            this.error(chalk.bold.red(e && e.message ? e.message : e), {exit: 2});
+        }
+
+        if (flags.i2fa) {
+            flags.code = await cli.prompt('2FA code', { type: 'hide' });
+        }
+        let ignoredFiles: Array<string>;
+        try {
+            ignoredFiles = await getIgnoredFiles(fd);
+        } catch (e) {
+            this.error(chalk.bold.red(e && e.message ? e.message : e));
+        }
+
+        const watcher = chokidar.watch(fd.folder, {
+            ignored: ignoredFiles,
+            awaitWriteFinish: true,
+            persistent: true,
+            interval: 300,
+        });
+        watcher
+            .on('change', async () => {
+                tasks(this, fd, flags)
+                .catch((e) => {
+                    this.log(chalk.bold.redBright(`   \u27ff  ${e && e.message ? e.message : e}`));
+                });
+            })
+            .on('ready', async () => {
+                tasks(this, fd, flags)
+                .catch((e) => {
+                    this.log(chalk.bold.redBright(`   \u27ff  ${e && e.message ? e.message : e}`));
+                });
+            });
+    }
+}
+const tasks = async (command: Command, fd: FolderDetails, flags: { [key: string]: any }): Promise<void> => {
+    let serverInfo: INormalLoginInfo | IPersonalAccessTokenLoginInfo | {};
+    let zipName;
+    try {
+        command.log('\n');
+        cli.action.start(chalk.bold.greenBright('   Checking App Report'));
+        checkReport(command, fd, flags);
+        cli.action.stop(chalk.bold.greenBright('\u2713'));
+
+        cli.action.start(chalk.bold.greenBright('   Packaging the app'));
+        zipName = await packageAndZip(command, fd);
+        cli.action.stop(chalk.bold.greenBright('\u2713'));
+
+        cli.action.start(chalk.bold.greenBright('   Getting Server Info'));
+        serverInfo = await getServerInfo(fd, flags);
+        cli.action.stop(chalk.bold.greenBright('\u2713'));
+
+        const status = await checkUpload({...flags, ...serverInfo}, fd);
+        if (status) {
+            cli.action.start(chalk.bold.greenBright('   Updating App'));
+            await uploadApp({...serverInfo, ...flags, update: true}, fd, zipName);
+            cli.action.stop(chalk.bold.greenBright('\u2713'));
+        } else {
+            cli.action.start(chalk.bold.greenBright('   Uploading App'));
+            await uploadApp({...serverInfo, ...flags}, fd, zipName);
+            cli.action.stop(chalk.bold.greenBright('\u2713'));
```

Comment:
Regarding there're more than one unicode symbols that appeared in the source code, I'd suggest using a "map" to store/retrieve them in one place. eg.

Use Object literal:
```javascript
const unicodeSymbols = {
    'checkMark': '\u2713',
    'heavyMultiplicationX': '\u2716',
    'longRightwardsSquiggleArrow': '\u27ff'
};
...
li.action.stop(chalk.red(unicodeSymbols['heavyMultiplicationX']));
```
Use Map:
```javascript
const unicodeSymbols = new Map([
    ['checkMark', '\u2713'],
    ['heavyMultiplicationX', '\u2716'],
    ['longRightwardsSquiggleArrow', '\u27ff']
]);
...
li.action.stop(chalk.red(unicodeSymbols.get('heavyMultiplicationX')));
```

Or as I suggested above, you can simply use ✓, ✖ and ⟿ in the Javascript string literal directly.